lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -12,11 +12,62 @@
 
 #include "circt/Dialect/Seq/SeqOps.h"
 #include "mlir/IR/Builders.h"
+#include "mlir/IR/DialectImplementation.h"
 
 using namespace mlir;
 using namespace circt;
 using namespace seq;
 
+ParseResult parseCompRegOp(OpAsmParser &parser, OperationState &result) {
+  llvm::SMLoc loc = parser.getCurrentLocation();
+  SmallVector<OpAsmParser::OperandType, 4> operands;
+  if (parser.parseOperandList(operands))
+    return failure();
+  switch (operands.size()) {
+  case 0:
+    return parser.emitError(loc, "expected operands");
+  case 1:
+    return parser.emitError(loc, "expected clock operand");
+  case 2:
+    // No reset.
+    break;
+  case 3:
+    return parser.emitError(loc, "expected resetValue operand");
+  case 4:
+    // reset and reset value included.
+    break;
+  default:
+    return parser.emitError(loc, "too many operands");
+  }
+
+  if (parser.parseOptionalAttrDict(result.attributes))
+    return failure();
+
+  if (parser.parseColon())
+    return failure();
+
+  Type ty;
+  if (parser.parseType(ty))
```

Comment:
Please merge all these together to reduce lines of code and make it more clear:

if (parseoptionalattrdict() ||
    parseColon||
   parseType )
  return failure